### PR TITLE
feat(API): add Custom Metadata endpoints

### PR DESCRIPTION
**Pull request by anthonyshchang**
https://github.com/phrase/openapi/assets/108525179/2a4151e2-f39c-49de-bf1d-b408d19ed7f6

### DONT MERGE
DONT MERGE UNTIL FEATURE TOGGLE IS REMOVED!

### Ticket
https://phrase.atlassian.net/browse/TSS-2511

### CODE REVIEW

**Review by forelabs — APPROVED** [review]

File: parameters.yaml [hunk 1]
```diff
@@ -31,6 +31,14 @@ account_id:
   required: true
   schema:
     type: string
+data_type:
```
Comment (jablan) [inline parameters.yaml:34]: I suggest renaming this to `custom_metadata_data_type` or `custom_metadata_type` to avoid confusion

File: paths.yaml [hunk 2]
```diff
@@ -47,6 +47,18 @@
     "$ref": "./paths/styleguides/update.yaml"
   delete:
     "$ref": "./paths/styleguides/destroy.yaml"
+"/accounts/{account_id}/custom_metadata/labels":
+  get:
+    "$ref": "./paths/custom_metadata_properties/index.yaml"
+"/accounts/{account_id}/custom_metadata/labels/{id}":
+  get:
+    "$ref": "./paths/custom_metadata_properties/show.yaml"
+  post:
```
Comment (jablan) [inline paths.yaml:56]: shouldn't POST be on a collection rather than on an item

File: paths/custom_metadata_properties/create.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
+---
+summary: Create a property
+description: Create a new custom metadata property.
+operationId: custom_metadata_property/create
+tags:
+- Custom Metadata
+parameters:
+- "$ref": "../../parameters.yaml#/X-PhraseApp-OTP"
+- "$ref": "../../parameters.yaml#/account_id"
+- "$ref": "../../parameters.yaml#/id"
```
Comment (jablan) [inline paths/custom_metadata_properties/create.yaml:10]: see above comment

File: paths.yaml [hunk 4]
```diff
@@ -47,6 +47,18 @@
     "$ref": "./paths/styleguides/update.yaml"
   delete:
     "$ref": "./paths/styleguides/destroy.yaml"
+"/accounts/{account_id}/custom_metadata/labels":
```
Comment (jablan) [inline paths.yaml:50]: so we decided to leave `labels` in the API rather than renaming to `properties`?

**Review by jablan — COMMENTED** [review]
I had some questions

File: paths/custom_metadata_properties/index.yaml [hunk 5]
```diff
@@ -10,7 +10,7 @@ tags:
 parameters:
 - "$ref": "../../parameters.yaml#/X-PhraseApp-OTP"
 - "$ref": "../../parameters.yaml#/account_id"
-- "$ref": "../../parameters.yaml#/data_type"
+- "$ref": "../../parameters.yaml#/custom_metadata_data_type"
```
Comment (jablan) [inline paths/custom_metadata_properties/index.yaml:13]: I'm curious, why is there a data_type parameter? is it for filtering properties only for the given type? what was the use case for adding this?

File: paths/custom_metadata_properties/index.yaml [hunk 6]
```diff
@@ -10,7 +10,7 @@ tags:
...
 - description: id of project that the properties belong to
   example:
   - abcd1234cdef1234abcd1234cdef1234
```
Comment (jablan) [inline paths/custom_metadata_properties/index.yaml:16]: why an array here?

Comment (jablan) [reply paths/custom_metadata_properties/index.yaml:16]: yeah, scalar would just be

```yaml
  example: abcd1234cdef1234abcd1234cdef1234
```

File: paths.yaml (hunk 4 above)
Comment (jablan) [reply paths.yaml:50]: so now needs to be updated here too, right?

**Review by malvarezphrase — APPROVED** [review]

**Comment by ildarsafin** [comment]
Tested the creation of key and response body: We are getting CM inside:

Screenshot 2023-12-13 at 10 25 18

File: parameters.yaml [hunk 7]
```diff
@@ -35,9 +35,18 @@ custom_metadata_data_type:
   in: query
   name: data_type
   description: Data Type of Custom Metadata Property
-  required: false
+  required: true
```
Comment (jablan) [inline parameters.yaml:38]: by doing this, it is also mandatory in the `index` call, right? I'm pretty sure we don't want that.

Comment (ildarsafin) [reply parameters.yaml]: this is actually confusing since it's rendered also in create/update CM pages and it's shown as not required which is not true

Comment (jablan) [reply parameters.yaml]: yeah, I'm afraid it can't be reused as a single reference then.

Comment (jablan) [reply parameters.yaml]: wait, perhaps it can be specified using `required` array

Comment (ildarsafin) [reply parameters.yaml]: which one?

Comment (jablan) [reply parameters.yaml]: nah, forget it. you can't reuse the same reference.

**Pull request merged**